packages/ui/react-ui-components/src/components/MarkdownViewer/MarkdownViewer.tsx

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+import React, { type PropsWithChildren } from 'react';
+import ReactMarkdown, { type Options as ReactMarkdownOptions } from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+
+import { type ThemedClassName } from '@dxos/react-ui';
+import { SyntaxHighlighter } from '@dxos/react-ui-syntax-highlighter';
+import { mx } from '@dxos/react-ui-theme';
+
+// TODO(burdon): Benchmark vs. codemirror (which would be more consistent).
+// TODO(burdon): Checkbox (input).
+
+export type MarkdownViewerProps = ThemedClassName<
+  PropsWithChildren<{
+    content?: string;
+    components?: ReactMarkdownOptions['components'];
+  }>
+>;
+
+/**
+ * Transforms markdown text into react elements.
+ * https://github.com/remarkjs/react-markdown
+ * markdown -> remark -> [mdast -> remark plugins] -> [hast -> rehype plugins] -> components -> react elements.
+ */
+export const MarkdownViewer = ({ classNames, children, components, content = '' }: MarkdownViewerProps) => {
+  return (
+    <div className={mx(classNames)}>
+      <ReactMarkdown remarkPlugins={[remarkGfm]} skipHtml components={{ ...defaultComponents, ...components }}>
+        {content}
+      </ReactMarkdown>
+      {children}
+    </div>
+  );
+};
+
+const defaultComponents: ReactMarkdownOptions['components'] = {
+  h1: ({ children }) => {
+    return <h1 className='pbs-1 pbe-1 text-xl'>{children}</h1>;
+  },
+  h2: ({ children }) => {
+    return <h2 className='pbs-1 pbe-1 text-lg'>{children}</h2>;
+  },
+  h3: ({ children }) => {
+    return <h3 className='pbs-1 pbe-1 text-base'>{children}</h3>;
+  },
+  blockquote: ({ children, ...props }) => (
+    <blockquote className='pis-4 mbs-2 mbe-2 pbs-2 pbe-2 border-l-4 border-primary-500 text-primary-500' {...props}>
```

Comment:
```suggestion
    <blockquote className='pis-4 mbs-2 mbe-2 pbs-2 pbe-2 border-l-4 border-accentSurface text-accentText' {...props}>
```

Please continue to use semantic tokens, physical tokens will become a source of design- and tech-debt.